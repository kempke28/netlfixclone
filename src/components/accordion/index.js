import React, {useState} from 'react';
import import { Container, Row, Column, Link, Title, Text, Break } from

export default function Accordion({children, ...restProps}) {
    return (
        <Container {...restProps}>
            <Inner>

            </Inner>
        </Container>
    )
};

Accordion.Title = function AccordionTitle({children, ...restProps}) {
    return <Title {...restProps}>{children}</Title>
}

Accordion.Frame = function AccordionFrame({children, ...restProps}) {
    return <Frame {...restProps}>{children}</Frame>
}

Accordion.Item = function AccordionItem({children, ...restProps}) {


    return <Item {...restProps}>{children}</Item>
}

Accordion.Frame = function AccordionFrame({children, ...restProps}) {
    return <Frame {...restProps}>{children}</Frame>
}