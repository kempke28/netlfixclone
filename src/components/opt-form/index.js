import React from 'react';
import {Container, Input, Button, Text, Break } from "./styles/opt-form"

export default function OptForm({children, ...restProps}) {
    return <Container {...restProps}>{children}</Container>;
};

OptForm.Input = function optFormInput({children, ...restProps}) {
    return <Input {...restProps} />;
}

OptForm.Button = function optFormutton({children, ...restProps}) {
    return (
        <Button {...restProps}>
            {children}
            <img src="/images/icons/chevron-right.png" alt="Try now"/>
        </Button>
    )
};

OptForm.Text = function OptFormText({children, ...restProps}) {
    return <Text {...restProps}>{children}</Text>;
}

OptForm.Break = function OptFormBreak({children, ...restProps}) {
    return <Break {...restProps} />;
}