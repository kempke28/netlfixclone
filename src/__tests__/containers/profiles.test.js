import React from 'react';
import { render } from '@testing-library/react';
import { Profiles } from '../../components';

describe('<Profiles />', () => {
    it('renders the <Profiles /> with populated data', () => {
        const { container, getByText, getByTestId } = render(
            <Profiles>
                <Profiles.Title>Whos is watching</Profiles.Title>
                <Profiles.List>
                    <Profiles.User onClick={() => setProfile({ displayName: user.displayName, photoURL: user.photoURL })}
                                   data-testid="user-profile">
                        <Profiles.Picture src='/images/dod.png' data-testid='profile-picture' />
                        <Profiles.Name>Bob</Profiles.Name>
                    </Profiles.User>
                </Profiles.List>
            </Profiles>
        );

        expect(getByTestId('profile-picture')).toBeTruthy();
        expect(getByText('Whos is watching')).toBeTruthy();
        expect(getByText('Bob')).toBeTruthy();
        expect(container.firstChild).toMatchSnapshot();
    });
});
