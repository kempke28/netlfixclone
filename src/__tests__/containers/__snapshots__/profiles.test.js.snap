// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profiles /> renders the <Profiles /> with populated data 1`] = `
<div
  class="Profiles__Container-sc-1ytbi8g-0 kurRtl"
>
  <h1
    class="Profiles__Title-sc-1ytbi8g-1 hFIgFh"
  >
    Whos is watching
  </h1>
  <ul
    class="Profiles__List-sc-1ytbi8g-2 dndWpG"
  >
    <li
      class="Profiles__Item-sc-1ytbi8g-5 fdNtbE"
      data-testid="user-profile"
    >
      <img
        class="Profiles__Picture-sc-1ytbi8g-4 jYuutZ"
        data-testid="profile-picture"
        src="/images/users//images/dod.png.png"
      />
      <p
        class="Profiles__Name-sc-1ytbi8g-3 bcVpUK"
      >
        Bob
      </p>
    </li>
  </ul>
</div>
`;
